services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/core/rabbitmq.py

```diff
@@ -0,0 +1,171 @@
+from __future__ import annotations
+
+import asyncio
+import json
+import logging
+import socket
+from asyncio import CancelledError, Queue, Task
+from typing import Any, Dict, List, Optional, Union
+
+import aio_pika
+import tenacity
+from fastapi import FastAPI
+from models_library.projects import ProjectID
+from models_library.projects_nodes import NodeID
+from models_library.users import UserID
+from servicelib.rabbitmq_utils import RabbitMQRetryPolicyUponInitialization
+from settings_library.rabbit import RabbitSettings
+
+from ..core.settings import DynamicSidecarSettings
+
+log = logging.getLogger(__file__)
+
+
+SLEEP_BETWEEN_SENDS: float = 1.0
+
+
+def _close_callback(sender: Any, exc: Optional[BaseException]) -> None:
+    if exc:
+        if isinstance(exc, CancelledError):
+            log.info("Rabbit connection was cancelled", exc_info=True)
+        else:
+            log.error(
+                "Rabbit connection closed with exception from %s:",
+                sender,
+                exc_info=True,
+            )
+
+
+def _channel_close_callback(sender: Any, exc: Optional[BaseException]) -> None:
+    if exc:
+        log.error(
+            "Rabbit channel closed with exception from %s:", sender, exc_info=True
+        )
+
+
+@tenacity.retry(**RabbitMQRetryPolicyUponInitialization().kwargs)
```

Comment:
since you do have a retry policy for this one. I guess you would need one for the docker registry as well

Reply:
will try to add one